active_learning/tool_tracking_al/active_learning.py

```diff
@@ -0,0 +1,573 @@
+# Copyright (c) MONAI Consortium
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Import Statements
+import os
+import argparse
+import copy
+import random
+import json
+import torch
+import time
+import pickle
+import numpy as np
+import matplotlib.pyplot as plt
+
+from monai.metrics import DiceMetric, MeanIoU
+from monai.utils import set_determinism
+from monai.losses import DiceLoss
+from monai.data import DataLoader, CacheDataset, Dataset, decollate_batch
+from monai.transforms import (
+    LoadImaged,
+    EnsureChannelFirstd,
+    CastToTyped,
+    EnsureTyped,
+    ScaleIntensityd,
+    RandRotated,
+    Lambdad,
+    RandZoomd,
+    Resized,
+    ToMetaTensord,
+    Compose,
+    AsDiscrete,
+)
+
+from torch.utils.tensorboard import SummaryWriter
+from monai.networks.nets import FlexibleUNet
+
+# Argument Parser
+parser = argparse.ArgumentParser(description="Active Learning Settings")
+
+# Directory & Json & Seed
+parser.add_argument("--base_dir", default="/home/vishwesh/experiments/robo_tool_experiments/variance_sanity", type=str)
+parser.add_argument("--data_root", default="/scratch_2/robo_tool_dataset_2023", type=str)
+parser.add_argument("--json_path", default="/scratch_2/robo_tool_dataset_2023/data_list.json", type=str)
+parser.add_argument("--seed", default=120, type=int)
+
+# Active learning parameters
+parser.add_argument("--active_iters", default=3, type=int)
+parser.add_argument("--dropout_ratio", default=0.2, type=float)
+parser.add_argument("--mc_number", default=10, type=int)
+parser.add_argument("--initial_pool", default=10, type=int)
+parser.add_argument("--queries", default=10, type=int)
+parser.add_argument("--strategy", default="variance", type=str)  # Options available variance, random
+
+# DL Hyper-parameters
+parser.add_argument("--steps", default=300, type=int)  # Default is 2000, this needs to be tweaked more
+parser.add_argument("--val_interval", default=1, type=int)
+parser.add_argument("--batch_size", default=1, type=int)
+parser.add_argument("--val_batch_size", default=1, type=int)
+parser.add_argument("--lr", default=1e-4, type=float)
```

Comment:
This also seems unreferenced in the code

Reply:
This is referenced now.